src/View/AbstractDirectoryView.vala

```diff
@@ -2566,8 +2568,10 @@ namespace FM {
                 selected_files.@foreach ((file) => {
                     var goffile = GOF.File.get_by_uri (file.get_display_target_uri ());
                     goffile.query_update ();
-                    action_files.append (goffile);
+                    action_files.prepend (goffile);
                 });
+
+                action_files.reverse ();
```

Comment:
Prepend followed by reverse () is equivalent to append () as action_files starts off empty.